flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorEventDispatcherImpl.java

```diff
@@ -83,41 +130,101 @@ public void registerEventHandler(OperatorID operator, OperatorEventHandler handl
         }
     }
 
-    Set<OperatorID> getRegisteredOperators() {
-        return handlers.keySet();
+    boolean isRegisteredOperator(OperatorID operatorId) {
+        return handlers.containsKey(operatorId) || gatewayMap.containsKey(operatorId);
```

Comment:
Can you explain the case where an `operatorId` is in `gatewayMap` but not in `handlers`?